### PR TITLE
Berry fixes

### PR DESCRIPTION
**Pull request by HLXII**
Some fixes:
 - Refactor Oak Item Purchase into the BoughtOakItem class.
 - Add Stats for Mutation/Replants
 - Add Protein to Sinnoh Berry Master
 - Add Mulch Multiplier Control

Edit:
Was trying to get this into the 0.6.0 update, but Red beat me to it >:(.
I've added updating the save for the BoughtOakItems

### CODE REVIEW

File: src/scripts/oakItems/OakItems.ts [hunk 1]
```diff
@@ -103,16 +98,6 @@ class OakItems implements Feature {
...
-
-        // Loading purchaseList for BoughtOakItems
-        const purchaseList = json['purchaseList'];
-        if (purchaseList == null) {
-            this.purchaseList = this.defaults.purchaseList.map((v) => ko.observable<boolean>(v));
-        } else {
-            (purchaseList as boolean[]).forEach((value: boolean, index: number) => {
-                this.purchaseList[index](value);
-            });
-        }
```
Comment (RedSparr0w) [inline src/scripts/oakItems/OakItems.ts:115]: With `0.6.0` just being published, I think we will need to add something to `Updates.ts` for this?

Comment (HLXII) [reply src/scripts/oakItems/OakItems.ts:115]: Just added one for 0.6.1

File: src/scripts/items/MulchItem.ts [hunk 2]
```diff
@@ -4,8 +4,8 @@
 class MulchItem extends Item {
     type: MulchType;
 
-    constructor(type: MulchType, basePrice: number, displayName: string) {
-        super(MulchType[type], basePrice, GameConstants.Currency.farmPoint, { multiplierDecreaser: MultiplierDecreaser.Berry }, displayName);
+    constructor(type: MulchType, basePrice: number, displayName: string, description) {
```
Comment (RedSparr0w) [inline src/scripts/items/MulchItem.ts:7]:
```suggestion
    constructor(type: MulchType, basePrice: number, displayName: string, description: string) {
```

File: src/scripts/farming/FarmController.ts [hunk 3]
```diff
@@ -107,6 +114,18 @@ class FarmController {
...
+    private static getAmount() {
+        return Number(this.multipliers[this.multIndex()].replace(/\D/g, '')) || Infinity;
+    }
+
+    public static incrementMultiplier() {
+        this.multIndex((this.multIndex() + 1) % this.multipliers.length);
+    }
+
+    public static decrementMultiplier() {
+        this.multIndex((((this.multIndex() - 1) % this.multipliers.length) + this.multipliers.length) % this.multipliers.length);
```
Comment (RedSparr0w) [inline src/scripts/farming/FarmController.ts:126]:
```suggestion
        this.multIndex((this.multIndex() + this.multipliers.length - 1) % this.multipliers.length);
```

File: src/scripts/farming/BerryDeal.ts [hunk 4]
```diff
@@ -146,6 +146,18 @@ class BerryDeal {
...
         ));
+        list.push(new BerryDeal(
+            [
+                this.randomBerry(fourthGen),
+                this.randomBerry(fifthGen),
+            ],
+            [
+                SeededRand.intBetween(20, 40),
+                SeededRand.intBetween(5, 20),
+            ],
```
Comment (RedSparr0w) [inline src/scripts/farming/BerryDeal.ts:157]: I feel like this might be a little bit too cheap 🤷

Comment (HLXII) [reply src/scripts/farming/BerryDeal.ts:157]: Feel free to nerf the price if you want. I just chose randomly

File: src/scripts/items/MulchItem.ts [hunk 5]
```diff
@@ -17,7 +17,7 @@ class MulchItem extends Item {
...
 }
 
-ItemList['Boost_Mulch']   = new MulchItem(MulchType.Boost_Mulch, 50, 'Boost Mulch');
-ItemList['Rich_Mulch']  = new MulchItem(MulchType.Rich_Mulch, 100, 'Rich Mulch');
-ItemList['Surprise_Mulch']  = new MulchItem(MulchType.Surprise_Mulch, 150, 'Surprise Mulch');
-ItemList['Amaze_Mulch'] = new MulchItem(MulchType.Amaze_Mulch, 200, 'Amaze Mulch');
+ItemList['Boost_Mulch']   = new MulchItem(MulchType.Boost_Mulch, 50, 'Boost Mulch', 'Increases Berry growth rate.');
+ItemList['Rich_Mulch']  = new MulchItem(MulchType.Rich_Mulch, 100, 'Rich Mulch', 'Increases Berry harvest rate.');
+ItemList['Surprise_Mulch']  = new MulchItem(MulchType.Surprise_Mulch, 150, 'Surprise Mulch', 'Increases Berry mutation rate.');
+ItemList['Amaze_Mulch'] = new MulchItem(MulchType.Amaze_Mulch, 200, 'Amaze Mulch', 'Increases all Berry effects.');
```
Comment (RedSparr0w) [inline src/scripts/items/MulchItem.ts:23]: Not related to this area of the code,
But it's probably also worth adding description to the shop Oak Items.

Comment (HLXII) [reply src/scripts/items/MulchItem.ts:23]: Yeah... I wanted to, but that would require duplicating the strings (or at least storing them somewhere separate). At this point in the code we don't have access to the OakItems yet, so we can't just call App.game.oakItems.itemList[SquirtBottle].description. The string would have to be stored somewhere static and used in both locations instead.

Comment (RedSparr0w) [reply src/scripts/items/MulchItem.ts:23]: Or couldn't we just make oak shop items automatically fetch the oak items description?

Comment (HLXII) [reply src/scripts/items/MulchItem.ts:23]: I'm not sure how I feel about adding some sort of race condition on initialization. Is there a way we can ensure that items will always load first?

Comment (RedSparr0w) [reply src/scripts/items/MulchItem.ts:23]: Only way I can think of right now is making them static then referencing from the item 🤔 😩

Comment (HLXII) [reply src/scripts/items/MulchItem.ts:23]: I guess it'll have to do until we do some kind of I18N refactor so multiple locations can share the same string.

Comment (RedSparr0w) [reply src/scripts/items/MulchItem.ts:23]: > until we do some kind of I18N refactor

I honestly don't think that's going to happen any time soon, if ever..

Comment (RedSparr0w) [reply src/scripts/items/MulchItem.ts:23]: I've just added a default description for all shop Oak Items.

**Review by RedSparr0w — APPROVED** [review]
Changes look good, thanks for the PR!

**Pull request merged**